### DISCUSSION TITLE
how should I specify http versions for hurl requests?

### DISCUSSION THREAD

**Discussion by chenrui333** [Orange-OpenSource/hurl#1854] [Q&A] [answered]
how should I specify http versions for hurl requests? does it support http/2 and http3? thanks!

**Comment by fabricereix** [comment, marked as answer]
it is not supported yet
but it will be #1155

**Reply by chenrui333** [reply]
thanks you, just subscribed that thread 🎉

**Comment by jcamiel** [comment]
Hi @chenrui333 

on master (Hurl version 4.2.0+) we've now HTTP options version (`--http1.0`, `--http1.1`, `--http2`or `--http3`), provided the underlying libcurl supports the queried HTTP version.

You can specify it on CLI:

```shell
$ hurl --http3 foo.hurl
```

Or per request in a Hurl file:

```hurl
GET https://foo.com
[Options]
http3: true
HTTP 200

GET https://foo.com
[Options]
http2: true
HTTP 200
```

I've you can test it, don't hesitate to send us feedbacks!

Cheers !
